src/implementation/js/node/int.ts

```diff
@@ -0,0 +1,196 @@
+import * as frontend from 'llparse-frontend';
+
+import { Compilation } from '../compilation';
+import { Node } from './base';
+
+export class Int extends Node<frontend.node.Int> {
+  public doBuild(out: string[]): void {
+    this.prologue(out);
+
+    switch (this.ref.bytes) {
+      case 1: {
+        if (this.ref.signed) {
+          this.readInt8(out);
+        } else {
+          this.readUInt8(out);
+        }
+        break;
+      }
+
+      case 2: {
+        if (this.ref.littleEndian) {
+          this.readUInt16LE(out);
+        } else {
+          this.readUInt16BE(out);
+        }
+        break;
+      }
+
+      case 3: {
+        if (this.ref.littleEndian) {
+          this.readUInt24LE(out);
+        } else {
+          this.readUInt24BE(out);
+        }
+        break;
+      }
+
+      case 4: {
+        if (this.ref.littleEndian) {
+          this.readUInt32LE(out);
+        } else {
+          this.readUInt32BE(out);
+        }
+        break;
+      }
+    }
+
+    this.tailTo(out, this.ref.otherwise!);
+  }
+
+  private readInt8(out: string[]) {
+    const ctx = this.compilation;
+    const index = ctx.stateField(this.ref.field);
+
+    out.push(`${index} = (${ctx.bufArg()}[${ctx.offArg()}] & 2 ** 7) * 0x1fffffe;`)
```

Comment:
Nitpick: Let's use `1 << 7` instead of `2 ** 7`. Mixing binary and floating point operators looks fishy.

I'm not sure that this line does what you want it to do, but I understand that this is work in progress. We can discuss it later.

Reply:
This is based on how Node.js `Buffer` also implements reading integers: https://github.com/nodejs/node/blob/f5512ff61ecb668c2f49b7c05d3227ef7aa5e85f/lib/internal/buffer.js#L416

I thought being consistent with core would make sense here? 🤔 I'm not sure what the pros or cons of switching to `1 << 7` would be. 🤷‍♂

Reply:
we could also just inline the result of the multiplication / shifting by changing this to:

```suggestion
    out.push(`${index} = (${ctx.bufArg()}[${ctx.offArg()}] & ${2 ** 7}) * 0x1fffffe;`);
```

What do you think?